earlydecoder/load_module.go

```diff
@@ -23,6 +23,7 @@ type decodedModule struct {
 	DataSources          map[string]*dataSource
 	Variables            map[string]*module.Variable
 	Outputs              map[string]*module.Output
+	ModuleCalls          map[string]*module.ModuleCall
```

Comment:
map is a reasonable structure for https://github.com/hashicorp/terraform-ls/issues/598 and https://github.com/hashicorp/terraform-ls/issues/725 but I'm guessing that unnamed modules may also be helpful to parse in the context of https://github.com/hashicorp/terraform-ls/issues/869 - so we might end up turning this map into slice.

One could make the same argument about entries within the `required_providers` block. 🤔 

Something we can revisit later...